src/array.rs

```
@@ -122,21 +137,68 @@ impl Array {
+    /// let values: &[f32] = &[1.0, 2.0, 3.0];
+    /// let indices = Array::new(Dim4::new(&[3, 1, 1, 1]), values, Aftype::F32).unwrap();
+    /// ```
+    #[allow(unused_mut)]
+    pub fn new_strided<T>(slice: &[T], offset: i64,
+                          dims: Dim4, strides: Dim4,
+                          aftype: Aftype) -> Result<Array, AfError> {
+        unsafe {
+            let mut temp: i64 = 0;
+            let err_val = af_create_strided_array(&mut temp as MutAfArray,
+                                                  slice.as_ptr() as *const c_void,
+                                                  offset as DimT,
+                                                  dims.ndims() as c_uint,
+                                                  dims.get().as_ptr() as * const c_longlong,
+                                                  strides.get().as_ptr() as * const c_longlong,
+                                                  aftype as uint8_t);
+            match err_val {
+                0 => Ok(Array {handle: temp}),
+                _ => Err(AfError::from(err_val)),
+            }
+        }
+    }
+
     /// Returns the backend of the Array
     ///
     /// # Return Values
     ///
     /// Returns an value of type `Backend` which indicates which backend
     /// was active when Array was created.
-    pub fn get_backend(&self) -> Backend {
+    pub fn get_backend(&self) -> Result<Backend, AfError> {
         unsafe {
             let mut ret_val: i32 = 0;
-            af_get_backend_id(&mut ret_val as *mut c_int, self.handle as AfArray);
-            match ret_val {
-                1 => Backend::AF_BACKEND_CPU,
-                2 => Backend::AF_BACKEND_CUDA,
-                3 => Backend::AF_BACKEND_OPENCL,
-                _ => Backend::AF_BACKEND_DEFAULT,
+            let err_val = af_get_backend_id(&mut ret_val as *mut c_int, self.handle as AfArray);
+            match (err_val, ret_val) {
+                (0, 1) => Ok(Backend::CPU),
```

Good way to squash the error check in. :+1: